internal/app.go

```diff
@@ -208,6 +208,10 @@ func updateSecret(secretFile string, newContent []byte) (bool, error) {
 // Do an atomic write to the file which prevents race conditions for a reader.
 // Don't worry about writer synchronization as there is only one writer to these files.
 func safeWrite(secretFile string, newContent []byte) error {
+	dirName := filepath.Dir(secretFile)
+	if err := os.MkdirAll(dirName, 0o750); err != nil {
```

Comment:
Do we need to hardcode 0750?

Reply:
Another question is this: when I've seen this octal, I started to wonder if the `0640` value below is valid?

So, I looked here: https://github.com/golang/proposal/blob/master/design/19308-number-literals.md#octal-literals
And it seems that Golang maintainers plan to do something pesky to `0` prefixed octals some time in the future; so maybe need to change it to a more verbose `0o` prefix too.

Reply:
i've reworked this to get the parent's mod and use that - 5e51e336d81294b18c852ac98d2e443490b92e75